js/reference.js

```diff
@@ -4014,6 +4014,83 @@
             return this._cascadingDeletedItems;
         },
 
+        /**
+         * If prefill object is defined and has the required attributes, will return
+         * a BulkCreateForeignKeyObject object with the necessary objects used for a association modal picker
+         *
+         * @type {ERMrest.BulkCreateForeignKeyObject}
+         */
+        get bulkCreateForeignKeyObject() {
+            if (this._bulkCreateForeignKeyObject === undefined) {
+                verify(false, 'Call "computeBulkCreateForeignKeyObject" with the prefill object first');
+            }
+            return this._bulkCreateForeignKeyObject;
+        },
+
+        /**
+         * Will compute and return a BulkCreateForeignKeyObject if:
+         *   - the prefillObject is defined
+         *   - there are only 2 foreign key columns for this table that are not system columns
+         *   - using the prefill object, we can determine the main column for prefilling and leaf column for bulk selection
+         *
+         * @param {Object} prefillObject computed prefill object from chaise
+         * @returns {BulkCreateForeignKeyObject}
+         */
+        computeBulkCreateForeignKeyObject: function (prefillObject) {
+            if (this._bulkCreateForeignKeyObject === undefined) {
+                if (!prefillObject) {
+                    this._bulkCreateForeignKeyObject = null;
+                } else {
+                    // ignore the fks that are simple and their constituent column is system col
+                    // TODO: composite FKs
+                    var nonSystemColumnFks = this.table.foreignKeys.all().filter(function(fk) {
+                        return !(fk.simple && module._systemColumns.indexOf(fk.colset.columns[0]) !== -1);
```

Comment:
I think I should modify this check to the following to check eliminate composite foreign keys:
```
return fk.simple &&  module._systemColumns.indexOf(fk.colset.columns[0]) === -1
```
The foreign key has to be simple and NOT a system column.